plugins/inputs/prometheus/kubernetes.go

```diff
@@ -98,15 +99,44 @@ func (p *Prometheus) watch(ctx context.Context, client *k8s.Client) error {
 
 			switch eventType {
 			case k8s.EventAdded:
-				registerPod(pod, p)
-			case k8s.EventDeleted:
-				unregisterPod(pod, p)
+				if podReady(pod.Status.GetContainerStatuses()) {
+					registerPod(pod, p)
+				}
 			case k8s.EventModified:
+				if pod.Metadata.GetDeletionTimestamp() != nil && podConditionReady(pod.Status.GetConditions()) {
+					unregisterPod(pod, p)
+				} else if podReady(pod.Status.GetContainerStatuses()) && podConditionReady(pod.Status.GetConditions()) {
```

Comment:
Can the pod be not ready when the containers are ready or vis-versa?  Why doesn't this match the unregister condition on line 106?  Do we always receive a modified event in the case that the pod is removed?

Reply:
I know when I didn't have them, a newly created pod would be registered, unregistered, and then registered again before it was "up"

Reply:
Not sure what I was doing to think they were different, they seem to be true at the same time. We do not always receive a modified event in the case that a pod is removed (external deletion/`docker rm`, etc...), but we always do when k8s requests the deletion.